### PR TITLE
ACN_CD_OktaAnalyticsRulesFix

### PR DESCRIPTION
**Pull request by chicduong**
Fixes #

## Proposed Changes

  - Corrected Syntax on Okta Analytics Rules
  -
  -

### CODE REVIEW

File: Detections/OktaSSO/PasswordSpray.yaml [hunk 1]
```diff
@@ -19,17 +19,17 @@ query: |
...
   | where TimeGenerated > timeframe
-  | where eventType_s =~ "user.account.lock"
-  | extend TimeGenerated = bin(TimeGenerated, 5m)
-  | join kind=inner LockOutEvents () on TimeGenerated
-  | summarize User = make_set(actor_alternateId_s) by ClientIP = client_ipAddress_s, City = client_geographicalContext_city_s, Country = client_geographicalContext_country_s, bin(TimeGenerated, 5m)
-  | sort by TimeGenerated desc
-  | extend timestamp = TimeGenerated, IPCustomEntity = ClientIP
+  | where eventType_s =~ "user.session.start"and outcome_reason_s in ("VERIFICATION_ERROR","INVALID_CREDENTIALS")
+  | summarize Users = make_set(actor_alternateId_s) by client_ipAddress_s, City = client_geographicalContext_city_s, Country = client_geographicalContext_country_s, bin(TimeGenerated, 5m)
+  | joinkind=inner FailedEvents () on client_ipAddress_s, TimeGenerated
```
Comment (aprakash13) [inline Detections/OktaSSO/PasswordSpray.yaml:32]: Space needed between join and kind. 
Might also be a good idea to put FailedEvents  in the bracket i.e. like below
| join kind=inner (FailedEvents)  on client_ipAddress_s, TimeGenerated

File: Detections/OktaSSO/PasswordSpray.yaml [hunk 2]
```diff
@@ -19,17 +19,17 @@ query: |
...
+  | sortby TimeGenerated desc
```
Comment (aprakash13) [inline Detections/OktaSSO/PasswordSpray.yaml:33]: Please give space between sort and by

File: Detections/OktaSSO/FailedLoginsFromUnknownOrInvalidUser.yaml [hunk 3]
```diff
@@ -21,14 +21,15 @@ query: |
...
+  | project client_ipAddress_s, actor_alternateId_s;
   Okta_CL
-  | where published_t > timeframe
-  | join kind=inner FailedLogins () on client_ipAddress_s
-  | where eventType_s =~ "user.session.start" and outcome_reason_s =~ "UNKNOWN_USER"
-  | summarize count() by ClientIP = client_ipAddress_s, City = client_geographicalContext_city_s, Country = client_geographicalContext_country_s, bin(TimeGenerated, 5m)
-  | sort by TimeGenerated desc
-  | extend timestamp = TimeGenerated, IPCustomEntity = ClientIP
+  | where TimeGenerated > timeframe
+  | join kind=inner FailedLogins () on client_ipAddress_s, actor_alternateId_s
```
Comment (aprakash13) [inline Detections/OktaSSO/FailedLoginsFromUnknownOrInvalidUser.yaml:30]: Might also be a good idea to put  FailedLogins in the bracket i.e. like below

| join kind=inner (FailedLogins) on client_ipAddress_s, actor_alternateId_s

File: Detections/OktaSSO/PasswordSpray.yaml (hunk 1 above)
Comment (chicduong) [reply Detections/OktaSSO/PasswordSpray.yaml]: revisions submitted

File: Detections/OktaSSO/PasswordSpray.yaml (hunk 2 above)
Comment (chicduong) [reply Detections/OktaSSO/PasswordSpray.yaml]: revisions submitted

File: Detections/OktaSSO/FailedLoginsFromUnknownOrInvalidUser.yaml (hunk 3 above)
Comment (chicduong) [reply Detections/OktaSSO/FailedLoginsFromUnknownOrInvalidUser.yaml]: revisions submitted

**Comment by chicduong** [comment]
Is there any updates / questions on the revisions?

**Review by shainw — APPROVED** [review]

**Review by aprakash13 — APPROVED** [review]
The changes look good. Ready to be merged.

**Pull request merged**